@here/harp-mapview/lib/image/ImageCache.ts

```diff
@@ -504,27 +247,26 @@ export class ImageCache {
     }
 
     /**
-     * Remove the cacheItem from cache and from the owners that have this item in their
-     * cache. If the item is used by another owner, the item is not removed, but the link
-     * to the owner is removed from the item, just like a reference count.
+     * Remove the cacheItem from cache, unless the item is used by another owner, in that case the
+     * link to the owner is removed from the item, just like a reference count.
      *
      * @param cacheItem The cache item to be removed.
-     * @param owner - Optional: Specify {@link ImageCacheOwner} removing the image.
+     * @param owner - Optional: Specify which owner ({@link any}) removes the image.
+     * If no owner is specified, the cache item is removed even if it has owners.
      */
-    private unlinkCacheItem(cacheItem: ImageCacheItem, owner?: ImageCacheOwner) {
+    private unlinkCacheItem(cacheItem: ImageCacheItem, owner?: any) {
         if (owner) {
             const ownerIndex = cacheItem.owners.indexOf(owner);
             if (ownerIndex >= 0) {
                 cacheItem.owners.splice(ownerIndex, 1);
             }
-            if (cacheItem.owners.length === 0) {
-                this.m_images.delete(cacheItem.imageItem.url);
-                this.cancelLoading(cacheItem.imageItem);
-                owner.imageRemoved(cacheItem.imageItem.url);
+            if (cacheItem.owners.length > 0) {
+                // Do not delete the item, it is still used.
+                return;
             }
-        } else {
-            this.m_images.delete(cacheItem.imageItem.url);
-            this.cancelLoading(cacheItem.imageItem);
         }
```

Comment:
The owners are no longer informed about removal of an image, owner.imageRemoved() is no longer being called. How do multiple MapViewImageCache instances (for shared caching of images for multiple MapViews) get the information that their images (same URL but different names) are no longer available in the cache?

Reply:
The owner is no longer optional, so I guess the images can no longer be removed until all owners have them removed individually. imageRemoved() is not required anymore.

Reply:
As we discussed, let's remove this callbacks as they shouldn't be needed if every owner is responsible for their own content.